src/components/credit-card-recommendation/credit-card-carousel.tsx

```diff
@@ -0,0 +1,127 @@
+
+  const cards = data?.cards.map((datum) => {
+    const { benefit, ...rest } = datum;
+
+    return rest;
+  });
+
+  const leftButtonClickCreatedCard = React.useRef<React.ReactNode>(null);
+  const rightButtonClickCreatedCard = React.useRef<React.ReactNode>(null);
+
+  const onLeftButtonClick = useThrottleTime(() => {
+    if (cards) {
+      leftButtonClickCreatedCard.current = null;
+      rightButtonClickCreatedCard.current = null;
+      const nextCardIndex = clamp(index - 1, cards.length);
+      leftButtonClickCreatedCard.current = (
+        <img
+          alt={cards[nextCardIndex]?.id}
+          src={cards[nextCardIndex]?.src}
+          className={[classes.Card, classes.EnterLeft].join(' ')}
+        />
+      );
+      setIndex(clamp(index - 1, cards.length));
+    }
+  }, timeOut);
+
+  const onRightButtonClick = useThrottleTime(() => {
+    if (cards) {
+      leftButtonClickCreatedCard.current = null;
+      rightButtonClickCreatedCard.current = null;
+      const nextCardIndex = clamp(index + 1, cards.length);
+      rightButtonClickCreatedCard.current = (
+        <img
+          alt={cards[nextCardIndex]?.id}
+          src={cards[nextCardIndex]?.src}
+          className={[classes.Card, classes.EnterRight].join(' ')}
+        />
+      );
+      setIndex(nextCardIndex);
+    }
+  }, timeOut);
+
+  // React.useEffect(() => {
+  //   onRightButtonClick();
+  // }, [autoPlayTrigger, onRightButtonClick]);
+
+  return (
+    <div className={classes.CarouselContainer}>
+      <div className={classes.Carousel}>
+        {cards?.map((item, i) => {
+          const indexLeft = clamp(index - 1, cards.length);
+          const indexRight = clamp(index + 1, cards.length);
+
+          if (i === indexLeft)
+            return (
+              <>
+                {leftButtonClickCreatedCard.current &&
+                  leftButtonClickCreatedCard.current}
+                <img
+                  key={item.id}
+                  className={[classes.Card, classes.Left].join(' ')}
+                  src={item.src}
+                  alt={item.id}
+                />
+              </>
+            );
+
+          if (i === indexRight)
+            return (
+              <>
+                {rightButtonClickCreatedCard.current &&
+                  rightButtonClickCreatedCard.current}
+                <img
+                  key={item.id}
+                  className={[classes.Card, classes.Right].join(' ')}
+                  src={item.src}
+                  alt={item.id}
+                />
+              </>
+            );
+
+          return (
+            <>
+              <img
+                key={item.id}
+                className={[classes.Card, classes.Active].join(' ')}
+                src={item.src}
+                alt={item.id}
+              />
+            </>
+          );
+        })}
+
+        <div className={classes.ButtonContainer}>
+          <button
+            onClick={onLeftButtonClick}
```

Comment:
```suggestion
            onClick={showNextCard}
```
`on*` 접두사는 관습적으로 이벤트 리스너를 의미하기 때문에 handler에 on을 붙히는 건 다소 어색하게 느껴질 수 있습니다. 그래서 토스에서는 `handle*` 네이밍을 많이 썼던 것으로 기억해요.
그리고 핸들러라고 해서 반드시 "자신이 호출 당하는 쪽의 관심사"를 이름에 표현해야할 필요가 없으니 함수가 가진 본연의 동작을 잘 표현할 수 있는 이름을 사용하는 것도 좋을 것 같아요!

Reply:
확실히 그렇네요! 좋은 코멘트 감사합니다 수정할게요